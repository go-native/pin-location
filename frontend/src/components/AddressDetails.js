import React from "react"

const AddressDetails = ({ address, onEdit, onRemove }) => {
  return (
    <div className="address_container">
      <div className="address_title">{address.title}</div>
      <div className="address_subtitle">{address.address}</div>
      <div className="address_subtitle">Latitude: {address.lat}</div>
      <div className="address_subtitle">Longitude: {address.lng}</div>
      <div className="address_actions">
        <button className="action-btn edit-btn" onClick={() => onEdit(address)}>
          Edit
        </button>
        <span className="separator">or</span>
        <button className="action-btn delete-btn" onClick={() => onRemove(address.id)}>
          Delete
        </button>
      </div>
    </div>
  );
};

export default AddressDetails
