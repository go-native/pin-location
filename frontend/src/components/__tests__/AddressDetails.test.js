import AddressDetails from "./../AddressDetails"
import React from "react"
import { shallow } from "enzyme"

const address = {
  id: "1544109127653",
  title: "Leipziger str.",
  address: "Leipziger strasse",
  lat: 52.511123,
  lng: 13.399605
}

describe("AddressDeatils", () => {
  it("should render properly", () => {
    const wrapper = shallow(<AddressDetails address={address} />)
    expect(wrapper).toMatchSnapshot()
  })

  it("should call proper function when edit button is clicked", () => {
    const onEditClickMock = jest.fn()
    const wrapper = shallow(
      <AddressDetails address={address} onEdit={onEditClickMock} />
    )
    const editBtnElem = wrapper.find(".edit-btn")
    editBtnElem.simulate("click")
    expect(onEditClickMock).toHaveBeenCalledTimes(1)
    expect(onEditClickMock).toHaveBeenCalledWith(address)
  })

  it("should call proper function when delete button is clicked", () => {
    const onRemoveClickMock = jest.fn()
    const wrapper = shallow(
      <AddressDetails address={address} onRemove={onRemoveClickMock} />
    )
    const editBtnElem = wrapper.find(".delete-btn")
    editBtnElem.simulate("click")
    expect(onRemoveClickMock).toHaveBeenCalledTimes(1)
    expect(onRemoveClickMock).toHaveBeenCalledWith(address.id)
  })
})
