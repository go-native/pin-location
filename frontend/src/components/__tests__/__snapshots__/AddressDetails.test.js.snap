// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressDeatils should render properly 1`] = `
<div
  className="address_container"
>
  <div
    className="address_title"
  >
    Leipziger str.
  </div>
  <div
    className="address_subtitle"
  >
    Leipziger strasse
  </div>
  <div
    className="address_subtitle"
  >
    Latitude: 
    52.511123
  </div>
  <div
    className="address_subtitle"
  >
    Longitude: 
    13.399605
  </div>
  <div
    className="address_actions"
  >
    <button
      className="action-btn edit-btn"
      onClick={[Function]}
    >
      Edit
    </button>
    <span
      className="separator"
    >
      or
    </span>
    <button
      className="action-btn delete-btn"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
